code/modules/xenoarchaeology/traits/xenoartifact_traits.dm

```diff
@@ -0,0 +1,123 @@
+			if(COMSIG_PARENT_ATTACKBY)
+				RegisterSignal(xenoa, COMSIG_PARENT_ATTACKBY, .proc/translate_attackby)
+			if(COMSIG_ITEM_ATTACK)
+				RegisterSignal(xenoa, COMSIG_ITEM_ATTACK, .proc/translate_attack)
+			if(COMSIG_MOVABLE_IMPACT)
+				RegisterSignal(xenoa, COMSIG_MOVABLE_IMPACT, .proc/translate_impact)
+			if(COMSIG_ITEM_AFTERATTACK)
+				RegisterSignal(xenoa, COMSIG_ITEM_AFTERATTACK, .proc/translate_afterattack)
+			if(COMSIG_ITEM_PICKUP)
+				RegisterSignal(xenoa, COMSIG_ITEM_PICKUP, .proc/translate_pickup)
+			if(COMSIG_ITEM_ATTACK_SELF)
+				RegisterSignal(xenoa, COMSIG_ITEM_ATTACK_SELF, .proc/translate_attack_self)
+			if(XENOA_SIGNAL)
+				RegisterSignal(xenoa, XENOA_SIGNAL, .proc/translate_attackby)
+	RegisterSignal(xenoa, XENOA_DEFAULT_SIGNAL, .proc/calculate_charge) //Signal sent by handles
+
+/datum/xenoartifact_trait/activator/Destroy(force, ...)
+	. = ..()
+	if(!xenoa)
+		return
+	for(var/s in signals)
+		UnregisterSignal(xenoa, s)
+	UnregisterSignal(xenoa, XENOA_DEFAULT_SIGNAL)
+	xenoa = null
+
+/datum/xenoartifact_trait/activator/proc/translate_attackby(datum/source, obj/item/thing, mob/user, params)
+	SEND_SIGNAL(xenoa, XENOA_DEFAULT_SIGNAL, thing, user, user)
+
+/datum/xenoartifact_trait/activator/proc/translate_attack_self(datum/source, mob/user, params)
+	SEND_SIGNAL(xenoa, XENOA_DEFAULT_SIGNAL, xenoa, user, user)
+
+/datum/xenoartifact_trait/activator/proc/translate_attack(mob/living/target, mob/living/user)
+	SEND_SIGNAL(xenoa, XENOA_DEFAULT_SIGNAL, xenoa, user, target)
+
+/datum/xenoartifact_trait/activator/proc/translate_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	SEND_SIGNAL(xenoa, XENOA_DEFAULT_SIGNAL, xenoa, hit_atom, throwingdatum) //Weird order to fix this becuase signals are mean
+
+/datum/xenoartifact_trait/activator/proc/translate_afterattack(atom/target, mob/user, params)
+	SEND_SIGNAL(xenoa, XENOA_DEFAULT_SIGNAL, target, params, user) //Weird order to fix this becuase signals are mean
+
+/datum/xenoartifact_trait/activator/proc/translate_pickup(mob/user, params)
+	SEND_SIGNAL(xenoa, XENOA_DEFAULT_SIGNAL, xenoa, params) //Weird order to fix this becuase signals are mean
+
+//End activator
+//Declare procs
+/datum/xenoartifact_trait/minor //Leave these here, for the future.
+
+/datum/xenoartifact_trait/major
+
+/datum/xenoartifact_trait/malfunction
+	flags = BLUESPACE_TRAIT | PLASMA_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/proc/activate(obj/item/xenoartifact/X, atom/target, atom/user, setup = TRUE) //Typical behaviour
+	return
+
+/datum/xenoartifact_trait/proc/on_item(obj/item/xenoartifact/X, atom/user, atom/item) //Item hint responses
+	return FALSE
+
+///This is better than initialize just for our specific control purposes, definitely not becuase I forgot to use it somehow.
+/datum/xenoartifact_trait/proc/on_init(obj/item/xenoartifact/X)
+	return
+
+/datum/xenoartifact_trait/proc/on_touch(obj/item/xenoartifact/X, atom/user) //Touch hint
+	return FALSE
```

Comment:
These abstract types should be declared before everything else, for readabilities sake